pkg/render/logstorage/linseed/linseed.go

```diff
@@ -299,6 +311,28 @@ func (l *linseed) linseedDeployment() *appsv1.Deployment {
 		l.cfg.KeyPair.VolumeMount(l.SupportedOSType()),
 	)
 
+	if l.cfg.ElasticClientSecret != nil {
+		// Add a volume for the required client certificate and key.
+		volumes = append(volumes, corev1.Volume{
+			Name: logstorage.ExternalCertsVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: logstorage.ExternalCertsSecret,
+				},
+			},
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      logstorage.ExternalCertsVolumeName,
+			MountPath: "/certs/elasticsearch/mtls",
+			ReadOnly:  true,
+		})
+
+		// Configure Linseed to use the mounted client certificate and key.
+		envVars = append(envVars, corev1.EnvVar{Name: "ELASTIC_MTLS_ENABLED", Value: "true"})
```

Comment:
Don;t we also need to override the ca path also?

Reply:
Nope, the CA remains the same.